egui/src/context.rs

```diff
@@ -813,19 +839,18 @@ impl Context {
 
         let platform_output: PlatformOutput = std::mem::take(&mut self.output());
 
-        let needs_repaint = if self.read().repaint_requests > 0 {
-            self.write().repaint_requests -= 1;
-            true
-        } else {
-            false
-        };
-        self.write().requested_repaint_last_frame = needs_repaint;
-
+        let repaint_after = self.read().repaint_after;
+        self.write().requested_repaint_last_frame = repaint_after.is_zero();
+        // make sure we reset the repaint_after duration.
+        // otherwise, if repaint_after is low, then any widget setting repaint_after next frame,
+        // will fail to overwrite the previous lower value. and thus, repaints will never
+        // go back to higher values.
```

Comment:
Is the `repaint_requests` logic gone, or just moved?

The current behavior is that calling `request_repaint` actually does _two_ repaints. That's because often you call it on an event, and then there are some thing in egui that has one frame of delay (e.g. window sizes are calculated one frame and then used to position the window the next frame). I'd like to keep that behavior! But only `request_repaint` (and maybe `request_repaint_after(Duration::ZERO)`) should do a double-repaint. `request_repaint_after(0.1s)` has no real reason to do so imho.

Reply:
hmm, its there in my local copy. let me check if i messed up with the commit smh

Reply:
its right there on line 842-244. I just removed the assignment of the `needs_repaint` bool. 

but i understand the issue now. I will instead set `repaint_after` to `Duration::ZERO` when the `repaint_requests` is greater than zero.